### PR TITLE
Add autosign_source, which may be used instead of autosign_content

### PR DESCRIPTION
**Pull request by downsj2**
autosign_content is nice, but very ugly if you only use hiera to pass parameters to classes.

This adds autosign_source, which can be used instead of autosign_content and does the obvious.

### CODE REVIEW

**Comment by downsj2** [comment]
I don't believe it's possible to write an rspec test which tests against content specified via the "source" attribute of a file resource, only against contents specified via "content".

File: spec/classes/puppet_server_config_spec.rb [hunk 1]
```diff
@@ -262,11 +262,24 @@
...
+              server           => true,
+              autosign_source  => 'puppet:///foo/bar',
+              autosign_entries => ['foo.bar'],
+           }"
+        end
+
+        it { should raise_error(Puppet::Error, /Cannot set both autosign_content\/autosign_source and autosign_entries/) }
+      end
+
+
```
Comment (mmoll) [inline spec/classes/puppet_server_config_spec.rb]: superfluous newline

**Comment by mmoll** [comment]
@downsj2 please rebase onto current master and squash the commits togetehr into one. Also, this needs a datatype instead of the validation.

**Comment by downsj2** [comment]
Replaced with PR #501, closing this one.

**Pull request closed without merging**